CppCoreGuidelines.md

```diff
@@ -20612,7 +20614,7 @@ and errors (when we didn't deal correctly with semi-constructed objects consiste
             : mx(check_size(x))
             , my(check_size(y))
             // now we know x and y have a valid size
-            , data(mx * my * sizeof(int)) // will throw std::bad_alloc on error
+            , data(mx * my) // will throw std::bad_alloc on error
```

Comment:
This example no longer matches the "bad" one before it. I don't think removing the `sizeof(int)` from only one of them is a good change. It will make people wonder if the reason this one is good is because it uses `int` not `char` (but that doesn't seem relevant to this rule).

Reply:
> This example no longer matches the "bad" one before it.

Yes, true. That was indeed my intention (to make this one "better" than the "bad" one, also in this respect – because apparently the picture has pixels whose type is `int`).

I have now changed also the "bad" example to use `int*`.

But I can of course just as well revert this whole change, if that sounds better?

Reply:
At least `char` sounds wrong to me (as per [SL.str.5](https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#Rstr-byte)). If not `int`, shouldn't the type be `byte`?

Shouldn't at least the "good" examples be "good" from all possible points of view, and not just the rule being discussed?

Reply:
I don't know what the "design" of the example is, so I don't know why it uses units of `sizeof(int)` but allocated as `char`. I agree it seems questionable. Maybe `uint8_t` would be better (because you can't do arithmetic on `byte` and many image processing transformations involve arithmetic).

Reply:
But it really seems that the pixel values are supposed to be `int`. So I think doing arithmetic on `uint8_t` looks very wrong, too.